repose-aggregator/components/filters/uri-stripper-filter/src/main/scala/org/openrepose/filters/uristripper/UriStripperFilter.scala

```diff
@@ -172,6 +240,25 @@ class UriStripperFilter @Inject()(configurationService: ConfigurationService)
     }
   }
 
+  private def removeFromPath(uri: String, tokenIndex: Int): String = {
+    logger.debug("Attempting to remove index: {} into uri: {}", tokenIndex.toString, uri)
+    Try(new URI(uri)) match {
+      case Success(parsedUri) =>
+        val pathTokens = splitPathIntoTokens(parsedUri.getRawPath)
+        Try(pathTokens.remove(tokenIndex)) match {
+          case Success(_) => //don't care
+          case Failure(e: IndexOutOfBoundsException) =>
+            logger.warn("Determined index outside of uri path length", e)
+            throw PathRewriteException("Determined index is out side the number of parsed path segments", e)
+          case Failure(e) => throw e
+        }
+        replacePath(parsedUri, joinTokensIntoPath(pathTokens))
```

Comment:
Mutable state confused me for a second. I thought about how you never caught the removal, but since this passes tests I'm assuming the the `splitPathIntoTokens` method returns a mutable collection of some sort.